### DISCUSSION TITLE
Question: Possible to use TMC2209s for all six axes?

### DISCUSSION THREAD

**Discussion by Tickolas** [JGMaker3dofficial/artistd#55] [Q&A]
Heya!

So I was planning to buy some other spare parts for the Artist-D, and had the idea to also replace the drivers with TMC2209s. Would you say it would be easy to modify the firmware to handle them and if so, how would i go about it? I've never edited firmware in that particular way before.

Thanks! =)

**Comment by mikejr83** [comment]
If there is enough interest I can build it on my personal repository. It is
fairly easy and more of figuring out the hardware jumpers.

**Reply by Tickolas** [reply]
Well, I'm interested for sure =) Also pretty sure it's going to be of use for the (I hope) budding Artist-D community, as soon as folks start switching out the jet turbine fans the whine from the 4988s is going to be much more noticeable. Also would be nice to be able to config linear advance to improve print quality.
Anything I can do to help?

**Reply by TwinkieXLII** [reply]
Linear advance is enabled

**Reply by mikejr83** [reply]
@Tickolas let me know when you're ready to test out a firmware. I just released my latest custom version and can easily post a bin that has been compiled for TMC2209's. I'll need to know if you're using them in UART or not.

**Reply by Tickolas** [reply]
> 
> 
> @Tickolas let me know when you're ready to test out a firmware. I just released my latest custom version and can easily post a bin that has been compiled for TMC2209's. I'll need to know if you're using them in UART or not.

Alright, have ordered some 2209's. Will let you know when they arrive, and I figure they should run in UART, right? Thanks! =)

**Reply by mikejr83** [reply]
I can compile the firmware changes necessary, but you'll have to determine if the jumper positions to put them in UART are correct. I have a guess on how they should be configured but ultimately you'll have to make that decision as I do not know if it is possible to damage stepper drivers with incorrect jumper configuration. I believe that they should be configured like the other Robin boards that MKS has but again, that is a guess as I have not found definitive documentation on the Robin Pro.
